Update handler.php

romanleitner opened the pull request:
GeneralUtility::array_merge_recursive_overrule is a depreciated method of 6.2, has been moved to ArrayUtility::mergeRecursiveWithOverrule. Thats why your operator is wrong, should be ">".

romanleitner committed:
Update handler.php

GeneralUtility::array_merge_recursive_overrule is a depreciated method of 6.2, has been moved to ArrayUtility::mergeRecursiveWithOverrule. Thats why your operator is wrong, should be ">".

tomasnorre commented:
@romanleitner 

Looks like we don't need the if at all. 
Both
\TYPO3\CMS\Core\Utility\GeneralUtility::explodeUrl2Array()
\TYPO3\CMS\Core\Utility\ArrayUtility::mergeRecursiveWithOverrule() 

are also available in TYPO3 6.2, so looks like we can skip the if completely.. 

https://api.typo3.org/typo3cms/62/html/class_t_y_p_o3_1_1_c_m_s_1_1_core_1_1_utility_1_1_general_utility.html#a88a1c7c48f48b670dae9d4e8e464e3d8
https://api.typo3.org/typo3cms/62/html/class_t_y_p_o3_1_1_c_m_s_1_1_core_1_1_utility_1_1_array_utility.html#a2eb6d2191036aeb191ef9911c98686ca

Please correct me if I'm wrong.

romanleitner commented:
Hi @tomasnorre ,

yes you are right. 
Methods
\TYPO3\CMS\Core\Utility\GeneralUtility::explodeUrl2Array()
\TYPO3\CMS\Core\Utility\ArrayUtility::mergeRecursiveWithOverrule()
are available in both versions 6.2.x and 7.6.x., so we can remove the condition. 

Its only about the method \TYPO3\CMS\Core\Utility\GeneralUtility::array_merge_recursive_overrule is not available anymore in v 7.6.x

lg
Roman

romanleitner committed:
Update Handler.php

Removed condition to check T3 version, not needed because methods ArrayUtility::mergeRecursiveWithOverrule and GeneralUtility::explodeUrl2Array are available in both T3 versions 6.2.x and 7.6.x.
